### PR TITLE
[MINOR][SQL] Remove signature from Hive thriftserver exception

### PR DESCRIPTION
**Pull request by srowen**
### What changes were proposed in this pull request?

Don't return expected signature to caller in Hive thriftserver exception 

### Why are the changes needed?

Please see private@ discussion

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master/3.5/3.4/3.3.

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/CookieSigner.java [hunk 1]
```diff
@@ -81,8 +81,7 @@ public String verifyAndExtract(String signedStr) {
       LOG.debug("Signature generated for " + rawValue + " inside verify is " + currentSignature);
     }
     if (!MessageDigest.isEqual(originalSignature.getBytes(), currentSignature.getBytes())) {
-      throw new IllegalArgumentException("Invalid sign, original = " + originalSignature +
-        " current = " + currentSignature);
+      throw new IllegalArgumentException("Invalid sign");
```
Comment (beliefer) [inline sql/hive-thriftserver/src/main/java/org/apache/hive/service/CookieSigner.java:84]: Could we add` incorrect signature`? `Invalid sign` makes confusion.

Comment (srowen) [reply sql/hive-thriftserver/src/main/java/org/apache/hive/service/CookieSigner.java:84]: I went for minimal change here. This is copied from Hive and meant to be a stopgap until Hive addresses an ongoing discussion about a potential security issue here

**Comment by LuciferYang** [comment]
late LGTM, hahaha, this is a very old security issue.

**Pull request closed without merging**